proto/common.proto

```diff
@@ -45,6 +61,7 @@ message WorkerNode {
     UNSPECIFIED = 0;
     STARTING = 1;
     RUNNING = 2;
+    CORDONED = 3;
```

Comment:
Yap. I am working on changing that and adding `is_schedulable` as a property, instead of a cordoned state.

Reply:
I switched to the `is_schedulable`, which I developed [here](https://github.com/risingwavelabs/risingwave/pull/10313). I also had a[ discussion](https://risingwave-labs.slack.com/archives/C05AR07Q82U/p1686756386851889) wether or not we want to negate this flag, i.e. use `is_unschedulable` or `is_cordoned` instead of `is_schedulable`. @BugenZhao You mentioned that we may need to consider backwards compatibility? Please let me know if I need to negate this flag.

Reply:
If the cluster is upgraded, all new fields will have the default values. For example, `is_schedulable` will by default be `false`. I'm not sure whether this can happen. cc @yezizp2012

Reply:
If the cluster is under rolling update, it will not only have backwards compatibility but also will cause unexpected issues if the meta and the compute node are in different upgrade orders. I suggest to use `is_unschedulable` instead. And there's also one thing need to be considered, the `is_unschedulable` status should not be changed when a node restarts and registers itself to meta after it get cordoned. The status must only be managed by the two new RPCs.

Reply:
Thank you very much for pointing this out @yezizp2012. I changed the logic to `is_unschedulable`. I also added tests that make sure that restarting does not change the schedulability of a node in https://github.com/risingwavelabs/risingwave/pull/10143/commits/5dd710d209b7e816c08f9ef0d51953c342b3d97e. Please let me know if this works for you.